src/dndbeyond/base/character.js

```diff
@@ -135,35 +135,62 @@ class Character extends CharacterBase {
-                    max_hp = parseInt($(item).find("div[class*='styles_number']").text());
+        const getHpValues = (container, selectors) => {
+            let hp = null, max_hp = null;
+            for (let item of container.find(selectors.item).toArray()) {
+                const label = $(item).find(selectors.label).text().trim();
+                if (label === "Current") {
+                    const number = $(item).find(selectors.current);
+                    if (number.length > 0) hp = parseInt(number.val() || number.text());
+                } else if (label === "Max") {
+                    max_hp = parseInt($(item).find(selectors.max).text());
                 }
             }
-            const temp_item = $(".ct-health-summary__hp-group--temp .ct-health-summary__hp-item--temp .ct-health-summary__hp-item-content, .ct-quick-info__health div[class*='styles_temp']");
-            if (temp_item.length > 0) {
-                // Can be hp-empty class instead;
-                temp_hp = parseInt(temp_item.find(".ct-health-summary__hp-number, button[class*='styles_valueButton']").text()) || 0;
-            } else {
+            return { hp, max_hp };
+        };
+        
+        const getTempHp = (container, selectors) => {
+            const tempItem = container.find(selectors.temp);
+            return tempItem.length > 0 ? parseInt(tempItem.find(selectors.tempValue).val() || tempItem.find(selectors.tempValue).text()) || 0 : null;
+        };
+        
+        const healthPane = $(".b20-health-manage-pane");
+        const quickAccessPane = $(".ct-health-summary__hp-group--primary, .ct-quick-info__health");
+        
+        const selectors = {
+            pane: {
+                item: "div[class*='styles_container'] div[class*='styles_innerContainer'] div[class*='styles_item']",
+                label: "label[class*='styles_label'], span[class*='styles_label']",
+                current: "input[class*='styles_input']",
+                max: "div[class*='styles_maxContainer']",
+                temp: "div[class*='styles_container'] div[class*='styles_temp']",
+                tempValue: "input[class*='styles_input']"
+            },
+            quickAccess: {
+                item: ".ct-health-summary__hp-item, div[class*='styles_container'] div[class*='styles_innerContainer'] div[class*='styles_item']",
+                label: "label[class*='styles_label'], span[class*='styles_label']",
+                current: "button[class*='styles_valueButton']",
+                max: "div[class*='styles_number']",
+                temp: ".ct-health-summary__hp-group--temp .ct-health-summary__hp-item--temp .ct-health-summary__hp-item-content, div[class*='styles_temp']",
+                tempValue: ".ct-health-summary__hp-number, button[class*='styles_valueButton'], input[class*='styles_input']"
+            }
+        };
+        
+        let hp = 0, max_hp = 0, temp_hp = 0;
+        if(healthPane || quickAccessPane){
```

Comment:
This will always be true, I think you want `if (healthPane.length > 0 || quickAccessPane.length > 0)`